cilium/CFP-32759-cilium-mtls-via-mtls-agent.md

```diff
@@ -0,0 +1,304 @@
+
+#### MVP Components
+1. Transparent Redirection
+1. Cilium xDS Server 
+1. mTLS via Transparent Proxy
+
+### MVP Scope
+
+Users can enable Cilium mTLS per workload. All traffic leaving a node from an mTLS-enabled source destined for an mTLS-enabled destination on a different node will be sent over mTLS.
+
+This proposal is scoped to the following traffic configurations:
+* Traffic between workloads on the same node will not be proxied (and therefore not be encrypted)
+* Cross-node traffic on a single cluster is tunneled over mTLS
+* Node egress traffic to non-mTLS-enabled destinations is not proxied
+
+Out of scope configurations (design should be extensible to support these configuration, but they are out of scope for this proposal):
+* Multi-cluster traffic
+
+### Enablement API
+
+For an MVP, the mTLS agent functionality will be enabled in Cilium via an environment variable in the cilium configuration - _mtlsAgent: enable_. The Cilium agent will identify the mTLS agent via a pod label - _cilium.io/redir-type=proxy_. Workloads are enabled for redirection and identified by the Cilium agent with the following label - _cilium.io/redir-type=workload_. For the MVP, the user will apply the labels per workload. A future goal is to support enabling workloads by namespace with support for exclusions. The Cilium agent will use these labels to identify which pods should be added to the BPF redirection map and to validate requests for workload configuration.
+
+### Installation
+
+The mTLS agent will be installed and managed separately from Cilium. The order of installation for the mTLS agent and Cilium is not significant. Cilium must be installed or configured with the mTLS agent functionality enabled. The mTLS agent will be deployed separately. Once the mTLS agent and cilium are configured for cilium mTLS, all existing workloads with existing _cilium.io/redir-type=workload_ labels will be enabled for redirection.
+
+### Workload Configuration
+
+The mTLS agent requires enabled workload identity configuration to obtain workload certificates and provide a mapping of identity to certificate for lookup when establishing mTLS. The Cilium Agent is responsible for obtaining and providing the  workload configuration to the mTLS agent via proto-over-xds transport. 
+
+To support this functionality, the Cilium agent only needs to implement a minimal part of the xDS API. Instead of building on top of existing xDS APIs, a custom API could be implemented on top of the xDS protocol. This could follow the pattern that Istio's ztunnel project uses for the custom [Workload Discovery Service](https://github.com/istio/ztunnel/blob/db0a74212c42c66b611a71a9613afb501074e257/proto/workload.proto). Supporting a custom API or utilizing Istio’s existing custom WDS would provide Cilium with a simpler and more flexible API compared with standard xDS API. An example configuration could look like the following:
+
+```markdown
```

Comment:
can we add more details on mtls agent request and response from cilium agent. so mtls agent requests spiffe identity for workload from cilium agent based on pod service account?